core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -250,6 +297,18 @@ public void validate() {
       if (!trimmedQuery.endsWith(";")) {
         queryToExecute = trimmedQuery + ';';
       }
+
+      if (authMechanism.equalsIgnoreCase("Password") && Strings.isNullOrEmpty(oracleServerPassword)) {
+        throw new IllegalArgumentException("Password cannot be empty");
+      } else if (authMechanism.equalsIgnoreCase("Private Key") && Strings.isNullOrEmpty(privateKey)) {
+        throw new IllegalArgumentException("Private Key cannot be empty");
+      }
+
+      if (!(authMechanism.equalsIgnoreCase("Password") || authMechanism.equalsIgnoreCase("Private Key"))) {
```

Comment:
same comments as above. reverse to use string literal first in the equals. Also, this condition should be moved prior to the previous two conditions to line 300